pymbar/mbar_solvers.py

```diff
@@ -145,16 +149,28 @@ def mbar_objective_and_gradient(u_kn, N_k, f_k):
     The gradient is equation C6 in the JCP MBAR paper; the objective
     function is its integral.
     """
-    u_kn, N_k, f_k = validate_inputs(u_kn, N_k, f_k)
 
     log_denominator_n = logsumexp(f_k - u_kn.T, b=N_k, axis=1)
     log_numerator_k = logsumexp(-log_denominator_n - u_kn, axis=1)
-    grad = -1 * N_k * (1.0 - np.exp(f_k + log_numerator_k))
+    grad = -1 * N_k * (1.0 - jnp.exp(f_k + log_numerator_k))
 
     obj = math.fsum(log_denominator_n) - N_k.dot(f_k)
 
     return obj, grad
 
+def jax_mbar_hessian(u_kn, N_k, f_k):
+
+    jNk = 1.0*N_k
```

Comment:
what's the point of 1.0*?

Reply:
N_k is an int, jit (or maybe Jax, I can't recall) complains if it's not converted to float first.  I wasn't able to find the best direct conversion to float function that made jit happy. so I just multiplied by 1.0 to do it automatically.